src/vm/dllimport.cpp

```diff
@@ -6656,27 +6736,37 @@ HINSTANCE NDirect::LoadLibraryModule(NDirectMethodDesc * pMD, LoadLibErrorTracke
     if (SString::_wcsicmp(wszLibName, MAIN_CLR_MODULE_NAME_W) == 0)
     {
         hmod = GetCLRModule();
+        if (hmod == NULL)
```

Comment:
Should this condition rather be `if (hmod != NULL)` ?

Reply:
Also, do we need the check for NULL at all? This should never be NULL.

Reply:
I am surprised that this is not causing things to fail. It looks like that this special path that checks for `MAIN_CLR_MODULE_NAME_W` is probably just dead code and can be just deleted.

Reply:
Yes, sorry, fixed it.

Reply:
Could you please rather just delete it?

Reply:
I've fixed the comparison for now. I'll remove the check along with the PAL registration cleanup in another upcoming PR.

Reply:
@janvorli  can you please confirm if this check is now defunct?  Thanks.
https://github.com/dotnet/coreclr/blob/master/src/vm/dllimport.cpp#L6657-L6667

Reply:
Yes, this is defunct. The few places that PInvoke into coreclr in CoreLib use the proper name (libcoreclr) and I expect we will get rid of those few places too at some point.

Reply:
Thanks for the explanation, I deleted the code.